modules/ws/src/main/java/org/mule/module/ws/consumer/WSConsumer.java

```diff
@@ -107,20 +109,49 @@ public MuleEvent process(MuleEvent event) throws MuleException
     }
 
 
-    private void initializeWsdlRetrievalStrategy()
+    private void initializeWsdlRetrievalStrategy() throws InitialisationException
     {
         HttpRequesterConfig httpRequesterConfig = config.getConnectorConfig();
+        String url = config.getWsdlLocation();
+
+        boolean isHttpRequester = verifyIsHttpRequester(url);
 
-        if (httpRequesterConfig == null || httpRequesterConfig.getProxyConfig() == null)
+        if (httpRequesterConfig == null && config.isUseConnectorToRetrieveWsdl())
         {
-            wsdlRetrieverStrategy = new URLRetrieverStrategy();
+            throw new InitialisationException(CoreMessages.createStaticMessage("The useConnectorToRetrieveWsdl option requires connectorConfig to work"),
+                    this);
+        }
+
+        if(config.isUseConnectorToRetrieveWsdl() && isHttpRequester)
+        {
+            wsdlRetrieverStrategy = new HttpRequesterWsdlRetrieverStrategy(url, httpRequesterConfig.getTlsContext(), httpRequesterConfig.getProxyConfig(), muleContext);
         }
         else
         {
-            wsdlRetrieverStrategy = new ProxyWsdlRetrieverStrategy(httpRequesterConfig.getTlsContext(), httpRequesterConfig.getProxyConfig(), muleContext);
+            wsdlRetrieverStrategy = new URLRetrieverStrategy(url);
         }
     }
 
+    private boolean verifyIsHttpRequester(String url) throws InitialisationException
+    {
+        boolean isHttpRequester = false;
+        URL urlProtocol = null;
+        urlProtocol = IOUtils.getResourceAsUrl(config.getWsdlLocation(), getClass());
```

Comment:
Is this necessary? Isn't just relevant to check for http and https? Anything else will fall in the URLRetrieverStrategy right?